File: JAVA_APPLICATION/app/src/main/java/com/example/myapplication/AccountActivity.java
package com.example.myapplication;


import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class AccountActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_account);

        // Retrieve the logged-in email from SharedPreferences
        SharedPreferences sharedPreferences = getSharedPreferences("UserSession", MODE_PRIVATE);
        String savedEmail = sharedPreferences.getString("user_email", "No email found");

        // Update the TextView with the email
        TextView emailTextView = findViewById(R.id.emailText);
        emailTextView.setText(savedEmail);

        Button resetpw_button = findViewById(R.id.resetPasswordBtn);
        resetpw_button.setOnClickListener(v -> {
            Intent intent = new Intent(this, ResetPassword.class);
            startActivity(intent);
        });
    }
}

File: app/src/main/java/com/example/city_cycle_app/AccountActivity.kt
package com.example.city_cycle_app

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import androidx.core.content.edit

class AccountActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_account)

        // Retrieve the logged-in email from SharedPreferences
        val sharedPreferences = getSharedPreferences("UserSession", MODE_PRIVATE)
        val savedEmail = sharedPreferences.getString("user_email", "No email found")

        // Update the TextView with the email
        val emailTextView = findViewById<TextView>(R.id.emailText)
        emailTextView.text = savedEmail
    }
}